src/Temporalio/Bridge/OptionsExtensions.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+
+namespace Temporalio.Bridge
+{
+    internal static class OptionsExtensions
+    {
+        private static readonly ByteArrayRef ClientName = ByteArrayRef.FromUTF8("temporal-dotnet");
+        private static readonly ByteArrayRef ClientVersion = ByteArrayRef.FromUTF8(
+            Temporalio.Runtime.Version
+        );
+        private static readonly ByteArrayRef SdkName = ByteArrayRef.FromUTF8("sdk-dotnet");
+
+        public static unsafe Interop.ClientOptions ToInteropOptions(
+            this Temporalio.Client.TemporalConnectionOptions options,
+            Scope scope
+        )
+        {
+            if (options.TargetHost == null)
```

Comment:
Check for empty string too?

Reply:
:+1: will update (C# has a nice `IsNullOrEmpty` extension for just this case)

Reply:
Actually I just realized, empty is an error here. We don't want to treat empty and null the same.

Reply:
Fixed